futures/src/lib.rs

```diff
@@ -192,62 +56,185 @@ macro_rules! try_ready {
     })
 }
 
-// TODO: task_local macro
+/// A macro to create a `static` of type `LocalKey`
+///
+/// This macro is intentionally similar to the `thread_local!`, and creates a
+/// `static` which has a `with` method to access the data on a task.
```

Comment:
`with` was changed to [`get_mut`](https://github.com/rust-lang-nursery/futures-rs/pull/784/files#diff-48e5eeeacb6dc31a3e5b5ed5134314caR98).